### PR TITLE
5: Implement PathPlanner

### PR DESCRIPTION
**Pull request by cjahner1**
Add automatic path following to the robot to enable it to navigate to known positions on the field

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -57,6 +59,7 @@ private void configureBindings()
       ));
 
       m_primaryController.a().onTrue(new SeedFieldRelative(m_Drivetrain));
+      m_primaryController.b().whileTrue(new PathPlannerAuto("Amp Auto"));
```
Comment (jasonsoohoo) [inline src/main/java/frc/robot/RobotContainer.java]: Leave binds outside of the main project until we decide them

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -12,28 +12,52 @@
...
+  private final SwerveRequest.ApplyChassisSpeeds m_autoRequest = new SwerveRequest.ApplyChassisSpeeds();
 
+  /** 
+    @brief Configure PathPlanner objects for automatic path following
+  */
+  private void configurePathPlanner()
+  {
+    //Determine the radius of the drivebase from module locations
+    double driveBaseRadius = 0;
+    for (var moduleLocation : m_moduleLocations) {
```
Comment (jasonsoohoo) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: brackets

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -72,7 +98,17 @@ public Command applyRequest(Supplier<SwerveRequest> requestSupplier)
     return run(() -> this.setControl(requestSupplier.get()));  
   }
 
+  /**
+   * @brief Gets the robots current ChassisSpeeds object
+   * @return The robot's chassis speeds object
+   */
+  public ChassisSpeeds getCurrentRobotChassisSpeeds()
+  {
+        return m_kinematics.toChassisSpeeds(getState().ModuleStates);
```
Comment (jasonsoohoo) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: extra tabs?

**Review by jasonsoohoo — CHANGES_REQUESTED** [review]
Needs changes

**Comment by jasonsoohoo** [comment]
Your branch hasn't updated to the latest develop.

**Comment by cjahner1** [comment]
Changes made and tested

**Comment by jasonsoohoo** [comment]
Branch is still not updated, conflicts exist

**Comment by jasonsoohoo** [comment]
Remove existing auto. Don't implement any autos until we know for sure we are going to use them. It's ok to test, but we should only add paths that are competition ready. Please update constants to reflect the current robot. Update driver base characteristics, update camera locations.

**Review by jasonsoohoo — CHANGES_REQUESTED** [review]
Please update project

**Pull request closed without merging**